src/main/java/com/limechain/runtime/RuntimeBuilder.java

```diff
@@ -1,62 +1,150 @@
 import com.limechain.runtime.version.RuntimeVersion;
-import com.limechain.runtime.version.scale.RuntimeVersionReader;
+import com.limechain.storage.KVRepository;
+import com.limechain.storage.crypto.KeyStore;
+import com.limechain.storage.offchain.OffchainStorages;
+import com.limechain.storage.offchain.OffchainStore;
+import com.limechain.storage.offchain.StorageKind;
+import com.limechain.trie.BlockTrieAccessor;
 import com.limechain.trie.decoded.Trie;
 import com.limechain.trie.decoded.TrieVerifier;
 import com.limechain.utils.LittleEndianUtils;
 import com.limechain.utils.StringUtils;
-import io.emeraldpay.polkaj.scale.ScaleCodecReader;
 import io.emeraldpay.polkaj.types.Hash256;
 import lombok.extern.java.Log;
+import org.jetbrains.annotations.Nullable;
 import org.wasmer.ImportObject;
 import org.wasmer.Imports;
+import org.wasmer.Instance;
+import org.wasmer.Memory;
 import org.wasmer.Module;
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
 import java.util.logging.Level;
 
+/**
+ * A builder for runtime instances by provided bytecode source.
+ * This is the public interface for constructing {@link Runtime} instances.
+ */
 @Log
 public class RuntimeBuilder {
-    private static final byte[] ZSTD_PREFIX = new byte[]{82, -68, 83, 118, 70, -37, -114, 5};
+    private static final byte[] ZSTD_PREFIX = new byte[] {82, -68, 83, 118, 70, -37, -114, 5};
     private static final int MAX_ZSTD_DECOMPRESSED_SIZE = 50 * 1024 * 1024;
-    public static final int DEFAULT_HEAP_PAGES = 2048;
-    private static final byte[] CODE_KEY_BYTES =
-            LittleEndianUtils.convertBytes(StringUtils.hexToBytes(StringUtils.toHex(":code")));
 
+    // TODO: Do we want this implicit logic for building with the spring context to reside here (within the Builder)
+    //  or delegate it somewhere else (user-site)?
     /**
-     * Builds and returns a Runtime object for executing WebAssembly code.
+     * Builds and returns a ready-to-execute `Runtime` with dependencies from the global Spring context.
      *
-     * @param code The WebAssembly bytecode.
-     * @return A Runtime object.
+     * @param code the runtime wasm bytecode
+     * @return a ready to execute `Runtime` instance
+     * @implNote Make sure the Spring context is properly initialized, as building the runtime this way relies on it.
      */
     public Runtime buildRuntime(byte[] code) {
+        var nodeRole = AppBean.getBean(HostConfig.class).getNodeRole();
+
+        var db = AppBean.getBean(KVRepository.class);
+        var localStorage = new OffchainStore(db, StorageKind.LOCAL);
+        var persistentStorage = new OffchainStore(db, StorageKind.PERSISTENT);
+        // TODO:
+        //  This shouldn't be like that: base storage (for offchain_index api) is not the same as persistent storage.
+        //  Figure out the right approach, when it becomes relevant.
+        var baseStorage = persistentStorage;
```

Comment:
That's an out-of-scope one, but I noticed it.

The TODO regards the second of two issues; the first one being that our `localStorage` is acutally not fork-aware (just writes in the db) as of right now.

We discussed this is not a high priority for now, but I still feel the need to persist the knowledge of these issues somewhere... for when it becomes a problem (if at all).

Reply:
The TODO remains, as this issue is considered a lower priority for now.